### PR TITLE
fix degen <-> wdegen

### PR DESCRIPTION
**Pull request by greg-schrammel**
Fixes APP-####

## What changed (plus any additional context for devs)

adds degen native asset address

## Screen recordings / screenshots

https://github.com/user-attachments/assets/9426ee83-698a-41b8-a783-3a9bb6732dce

### CODE REVIEW

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/ed0631aa7df134666637f50f46abbf9196db3a1f.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/ed0631aa7df134666637f50f46abbf9196db3a1f.ipa) for ed0631aa7df134666637f50f46abbf9196db3a1f

File: src/references/index.ts [hunk 1]
```diff
@@ -217,6 +218,7 @@ export const NATIVE_ASSETS_MAP_PER_CHAIN: Record<ChainId, AddressOrEth> = {
   [ChainId.sepolia]: ETH_ADDRESS,
   [ChainId.zora]: ETH_ADDRESS,
   [ChainId.zoraSepolia]: ETH_ADDRESS,
+  [ChainId.degen]: AddressZero,
```
Comment (walmat) [inline src/references/index.ts:221]: for consistency can we use `ETH_ADDRESS` here. Don't think it's functionally any different but we're using it for other chains so would like to keep it consistent.

Comment (greg-schrammel) [reply src/references/index.ts:221]: just noticed this `NATIVE_ASSETS_MAP_PER_CHAIN` is not used anywhere, what was it for? 

in `NATIVE_ASSETS_PER_CHAIN` it gonna be `AddressZero` to work right

Comment (walmat) [reply src/references/index.ts:221]: I think it was moved to the swaps sdk iirc. Maybe NATIVE_ASSETS_PER_CHAIN was too?

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/e7e42fac2656496dd81c1d0ae91618ce2c99f744.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/e7e42fac2656496dd81c1d0ae91618ce2c99f744.ipa) for e7e42fac2656496dd81c1d0ae91618ce2c99f744

File: src/references/index.ts [hunk 2]
```diff
@@ -184,39 +184,7 @@ export const NATIVE_ASSETS_PER_CHAIN: Record<ChainId, AddressOrEth> = {
...
-  [ChainId.polygon]: MATIC_MAINNET_ADDRESS,
-  [ChainId.polygonAmoy]: MATIC_MAINNET_ADDRESS,
-  [ChainId.polygonMumbai]: MATIC_MAINNET_ADDRESS,
-  [ChainId.polygonZkEvm]: MATIC_MAINNET_ADDRESS,
-  [ChainId.rari]: ETH_ADDRESS,
-  [ChainId.scroll]: ETH_ADDRESS,
-  [ChainId.sepolia]: ETH_ADDRESS,
-  [ChainId.zora]: ETH_ADDRESS,
-  [ChainId.zoraSepolia]: ETH_ADDRESS,
+  [ChainId.degen]: AddressZero,
```
Comment (benisgold) [inline src/references/index.ts:187]: how come all the others are cast as `Address`

Comment (greg-schrammel) [reply src/references/index.ts:187]: idk why but they don't need to be

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/545652544c5e4623397d7f1b87f0954752c8db73.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/545652544c5e4623397d7f1b87f0954752c8db73.ipa) for 545652544c5e4623397d7f1b87f0954752c8db73

**Review by benisgold — APPROVED** [review]

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/6e8752f1a84afc6276639bd858bead9b2e5120a0.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/6e8752f1a84afc6276639bd858bead9b2e5120a0.ipa) for 6e8752f1a84afc6276639bd858bead9b2e5120a0

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/fdfe974514e0a52ed16e9c6cda6f1921c52c7ac6.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/gregs/app-1839-no-route-found-wdegen-to-degen/fdfe974514e0a52ed16e9c6cda6f1921c52c7ac6.ipa) for fdfe974514e0a52ed16e9c6cda6f1921c52c7ac6

**Review by walmat — APPROVED** [review]

**Pull request merged**